modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,20 +50,36 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeMajorityPeerIdSet: Option[NonEmptySet[PeerId]]
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
 
       private val logger = Slf4jLogger.getLogger[F]
+      private val maybeMajorityPeerIds = maybeMajorityPeerIdSet.map(_.toNonEmptyList)
 
       def pullLatestSnapshot: F[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] =
         globalL0ClusterStorage.getRandomPeer.flatMap { l0Peer =>
           l0GlobalSnapshotClient.getLatest(l0Peer).flatMap {
-            case ((snapshot, state)) =>
+            case (snapshot, state) =>
               snapshot.toHashedWithSignatureCheck.flatMap(_.liftTo[F]).map((_, state))
           }
         }
 
+      def pullLatestSnapshotFromMajority: F[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] =
+        maybeMajorityPeerIds match {
+          case None => pullLatestSnapshot
+          case Some(majorityPeerIds) =>
```

Comment:
IMO if there is no majority, the function `pullLatestSnapshotFromMajority` should fail that it cannot fetch the latest snapshot from majority

Reply:
Yeah, I think maciej discussed this earlier and I ended up agreeing with him -- since we don't have any retry mechanism or otherwise mechanism to detect if the priority peers are completely dead, we can't distinguish at all the case between an expected failure (i.e. something we want to fail on,) vs. an unexpected one. Meaning we'd need a hard failure on failure to detect majority. I believe this falls under the same scope of discussion as prior issue?